versioned_docs/version-v3.0/framework-integration/react.md

```diff
@@ -12,229 +12,506 @@ contributors:
+
+function App() {
+  return (
+    <div className="App">
+      <MyComponent first="Your" last="Name" />
+    </div>
+  );
+}
+
+export default App;
 ```
 
-With that, your component is now available to be used like any other React component.
+## API
 
-## FAQ's
+### componentCorePackage
 
-### What is the best format to write event names?
+**Optional**
 
-Event names shouldn’t include special characters when initially written in Stencil. Try to lean on using camelCased event names for interoperability between frameworks.
+**Default: The `components.d.ts` file in the Stencil project's `package.json` types field**
 
-### How do I add IE11 or Edge support?
+**Type: `string`**
 
-:::info
-As of Stencil v3, legacy browser support is deprecated, and will be removed in a future major version of Stencil.
-:::
+The name of the Stencil package where components are available for consumers (i.e. the value of the `name` property in your Stencil component library's `package.json`).
+This is used during compilation to write the correct imports for components.
 
-If you want your custom elements to be able to work on older browsers, you should add the `applyPolyfills()` that surround the `defineCustomElements()` function.
+For a starter Stencil project generated by running:
 
-```tsx
-import { applyPolyfills, defineCustomElements } from 'test-components/loader';
+```bash npm2yarn
+npm init stencil component my-component-lib
+```
 
-applyPolyfills().then(() => {
-  defineCustomElements();
-});
+The `componentCorePackage` would be set to:
+
+```ts
+// stencil.config.ts
+
+export const config: Config = {
+  ...,
+  outputTargets: [
+    reactOutputTarget({
+      componentCorePackage: 'my-component-lib',
+      // ... additional config options
+    })
+  ]
+}
 ```
 
-## API
+Which would result in an import path like:
+
+```js
+import { defineCustomElement as defineMyComponent } from 'my-component-lib/components/my-component.js';
+```
+
+:::note
+Although this field is optional, it is _highly_ recommended that it always be defined to avoid potential issues with paths not being generated correctly
+when combining other API arguments.
+:::
+
+### customElementsDir
+
+**Optional**
+
+**Default: 'dist/components'**
+
+**Type: `string`**
+
+If [includeImportCustomElements](#includeimportcustomelements) is `true`, this option can be used to specify the directory where the generated
+custom elements live. This value only needs to be set if the `dir` field on the `dist-custom-elements` output target was set to something other than
+the default directory.
+
+### excludeComponents
+
+**Optional**
+
+**Default: `[]`**
+
+**Type: `string[]`**
+
+This lets you specify component tag names for which you don't want to generate React wrapper components. This is useful if you need to write framework-specific versions of components. For instance, in Ionic Framework, this is used for routing components - like tabs - so that
+Ionic Framework can integrate better with React's Router.
```

Comment:
I think this should be 'React Router', React doesn't have a single routing solution (or a 'blessed' one):

```suggestion
Ionic Framework can integrate better with React Router.
```